homeassistant/components/velux/__init__.py

```diff
@@ -2,31 +2,35 @@
 import logging
 
 from pyvlx import PyVLX, PyVLXException
-import voluptuous as vol
 
+from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import CONF_HOST, CONF_PASSWORD, EVENT_HOMEASSISTANT_STOP
-from homeassistant.helpers import discovery
-import homeassistant.helpers.config_validation as cv
+from homeassistant.core import HomeAssistant
 
 DOMAIN = "velux"
 DATA_VELUX = "data_velux"
 PLATFORMS = ["cover", "scene"]
 _LOGGER = logging.getLogger(__name__)
 
-CONFIG_SCHEMA = vol.Schema(
-    {
-        DOMAIN: vol.Schema(
-            {vol.Required(CONF_HOST): cv.string, vol.Required(CONF_PASSWORD): cv.string}
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Component setup, run import config flow for each entry in config."""
+    if DOMAIN not in config:
+        return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config[DOMAIN]
         )
-    },
-    extra=vol.ALLOW_EXTRA,
-)
+    )
+
+    return True
 
 
-async def async_setup(hass, config):
-    """Set up the velux component."""
+async def async_setup_entry(hass: HomeAssistant, config_entry: ConfigEntry):
+    """Set up Velux using config flow."""
     try:
-        hass.data[DATA_VELUX] = VeluxModule(hass, config[DOMAIN])
+        hass.data[DATA_VELUX] = VeluxModule(hass, config_entry)
```

Comment:
I am not sure if this is best practice. Why not store it under the domain and entry id? (this will also allow users to set up multiple Velux config entries.


```python3
hass.data.setdefault(DOMAIN, {})
hass.data[DOMAIN][entry.entry_id] = {
        "velux_module": VeluxModule(hass, config_entry)
}

Reply:
This has been resolved. We have a VeluxModule per config entry now. Also cleaned up some other stuff.